lib/src/models/violation_level.dart

```diff
@@ -21,4 +21,15 @@ class ViolationLevel {
 
   @override
   String toString() => _name;
+
+  @override
+  int compareTo(ViolationLevel other) {
+    final valuesList = values.toList(growable: false);
+    return valuesList.indexOf(this).compareTo(valuesList.indexOf(other));
+  }
+
+  bool operator >(ViolationLevel other) => compareTo(other) > 0;
+  bool operator >=(ViolationLevel other) => compareTo(other) >= 0;
+  bool operator <(ViolationLevel other) => compareTo(other) < 0;
+  bool operator <=(ViolationLevel other) => compareTo(other) <= 0;
```

Comment:
@Anrock after this change, we need to be refactor UtilitySelector. functionViolationLevel